lib/compat/wordpress-6.7/class-gutenberg-rest-posts-controller-6-7.php

```diff
@@ -0,0 +1,700 @@
+		$args['date_query'] = array();
+
+		if ( isset( $registered['before'], $request['before'] ) ) {
+			$args['date_query'][] = array(
+				'before' => $request['before'],
+				'column' => 'post_date',
+			);
+		}
+
+		if ( isset( $registered['modified_before'], $request['modified_before'] ) ) {
+			$args['date_query'][] = array(
+				'before' => $request['modified_before'],
+				'column' => 'post_modified',
+			);
+		}
+
+		if ( isset( $registered['after'], $request['after'] ) ) {
+			$args['date_query'][] = array(
+				'after'  => $request['after'],
+				'column' => 'post_date',
+			);
+		}
+
+		if ( isset( $registered['modified_after'], $request['modified_after'] ) ) {
+			$args['date_query'][] = array(
+				'after'  => $request['modified_after'],
+				'column' => 'post_modified',
+			);
+		}
+
+		// Ensure our per_page parameter overrides any provided posts_per_page filter.
+		if ( isset( $registered['per_page'] ) ) {
+			$args['posts_per_page'] = $request['per_page'];
+		}
+
+		if ( isset( $registered['sticky'], $request['sticky'] ) ) {
+			$sticky_posts = get_option( 'sticky_posts', array() );
+			if ( ! is_array( $sticky_posts ) ) {
+				$sticky_posts = array();
+			}
+			if ( $request['sticky'] ) {
+				/*
+				 * As post__in will be used to only get sticky posts,
+				 * we have to support the case where post__in was already
+				 * specified.
+				 */
+				$args['post__in'] = $args['post__in'] ? array_intersect( $sticky_posts, $args['post__in'] ) : $sticky_posts;
+
+				/*
+				 * If we intersected, but there are no post IDs in common,
+				 * WP_Query won't return "no posts" for post__in = array()
+				 * so we have to fake it a bit.
+				 */
+				if ( ! $args['post__in'] ) {
+					$args['post__in'] = array( 0 );
+				}
+			} elseif ( $sticky_posts ) {
+				/*
+				 * As post___not_in will be used to only get posts that
+				 * are not sticky, we have to support the case where post__not_in
+				 * was already specified.
+				 */
+				$args['post__not_in'] = array_merge( $args['post__not_in'], $sticky_posts );
+			}
+		}
+
+		$args = $this->prepare_tax_query( $args, $request );
+
+		if ( ! empty( $request['format'] ) ) {
+			// If format is not an array, convert it to an array so that the
+			// required prefix can be added to all items.
+			$formats = is_array( $request['format'] ) ? $request['format'] : array( $request['format'] );
+
+			$tax_query = array( 'relation' => 'OR' );
+
+			// The default post format, 'standard', is not stored in the database.
+			// If 'standard' is part of the request, the query needs to exclude all post items that
+			// have a format assigned.
+			if ( in_array( 'standard', $formats, true ) ) {
+				$tax_query[] = array(
+					'taxonomy' => 'post_format',
+					'field'    => 'slug',
+					'terms'    => array(),
+					'operator' => 'NOT EXISTS',
+				);
+				// Remove the standard format, since it cannot be queried.
+				unset( $formats[ array_search( 'standard', $formats, true ) ] );
```

Comment:
I'm afraid I confused things a bit when I first misunderstood the data type of `$formats`. Given your response, I meant to indicate that your first code was fine with `array_diff()`.

Now, this is also good because we know `standard` is in the array, but otherwise in line 177 we would risk accessing `$formats[ false ]`, which is apparently benign but confusing still.

It's fine to leave this as-is, or revert to your original proposal.

A third option would be to make this entire block aware of the fact that `standard` is at a specific place, but I don't think this is necessary or blocking.

```php
$standard_at = array_search( 'standard', $formats, true );
if( false !== $standard_at ) {
	$tax_query[] = array( … );
	unset( $formats[ $standard_at ] );
}
```

though in review I realize there's another possibility not taken into account, which is that `standard` could appear many times. this seems like an unlikely situation, but if having `standard` in there is a problem, it might be worth considering removing all standard formats in the list. this could be done here, or in a follow-up patch.

at this point, an array filter or `foreach` might be the easier approach for PHP

```php
$has_seen_standard = false;
foreach ( $formats as $index => $format ) {
	if ( 'standard' !== $format ) {
		continue;
	}

	if ( ! $has_seen_standard ) {
		$has_seen_standard = true;
		$tax_query[] = array( … );
	}

	unset( $formats[ $index ] );
}
```

Reply:
Handling `standard` multiple times shouldn't be necessary. We can specify the `uniqueItems` keyword on the `format` schema definition.